pd/src/wallet/server.rs

```diff
@@ -1,22 +1,21 @@
 mod light;
 mod thin;
 
-use crate::state;
-
-impl state::Reader {
-    /// Checks a provided chain_id against the chain state.
-    ///
-    /// Passes through if the provided chain_id is empty or matches, and
-    /// otherwise errors.
-    fn check_chain_id(&self, provided: &str) -> Result<(), tonic::Status> {
-        if provided.is_empty() || self.chain_params_rx().borrow().chain_id == provided {
-            Ok(())
-        } else {
-            Err(tonic::Status::failed_precondition(format!(
-                "provided chain_id {} does not match chain_id {}",
-                provided,
-                self.chain_params_rx().borrow().chain_id
-            )))
-        }
-    }
-}
+// TODO: implement for WriteOverlay
+// impl state::Reader {
+//     /// Checks a provided chain_id against the chain state.
+//     ///
+//     /// Passes through if the provided chain_id is empty or matches, and
+//     /// otherwise errors.
+//     fn check_chain_id(&self, provided: &str) -> Result<(), tonic::Status> {
+//         if provided.is_empty() || self.chain_params_rx().borrow().chain_id == provided {
+//             Ok(())
+//         } else {
+//             Err(tonic::Status::failed_precondition(format!(
+//                 "provided chain_id {} does not match chain_id {}",
+//                 provided,
+//                 self.chain_params_rx().borrow().chain_id
+//             )))
+//         }
+//     }
+// }
```

Comment:
This is already implemented, I think as part of the `app::View` trait, so it can be deleted here as well. 

In fact the whole `wallet` module subtree can be deleted, since the new RPC implementation lives under `info` (in hope of some future unification with the ABCI query interface).